autoware_lanelet2_extension/include/autoware_lanelet2_extension/utility/utilities.hpp

```diff
@@ -86,6 +86,14 @@ double getLaneletLength3d(const lanelet::ConstLanelets & lanelet_sequence);
 lanelet::ArcCoordinates getArcCoordinates(
   const lanelet::ConstLanelets & lanelet_sequence, const geometry_msgs::msg::Pose & pose);
 
+/**
+ * @brief  This function uses custom centerline for the lanelet if exists defined with the attribute
+ * "waypoints" instead of centerline associated to the lanelet
+ */
+lanelet::ArcCoordinates getArcCoordinatesConsideringWaypoints(
```

Comment:
How about the following function name and the explanation?
```suggestion
/**
 * @brief  This function uses the centerline for the ego to follow.
 * - when the `use_waypoints` in the autoware_map_loader is true, 
 *   - the centerline in the lanelet2::LaneletMapPtr is used.
 * - when the `use_waypoints` in the autoware_map_loader is false, 
 *   - the waypoints tag in the lanelet2::LaneletMapPtr is used instead of the centerline.
 */
lanelet::ArcCoordinates getArcCoordinatesOnEgoCenterline(
```

Reply:
@takayuki5168 Thank you!

Function name suggestion look fine to me.

If I am not confused `use_waypoints` usage in your comment suggestion should be vice versa, right?

Reply:
@takayuki5168 I have applied the changes with the latest commit.